project_status_report/models/project_status.py

```diff
@@ -0,0 +1,172 @@
+    cost_color = fields.Char(readonly=True,
+                             default='#00ff00',
+                             states={'draft': [('readonly', False)]})
+    cost_remarks = fields.Html(readonly=True,
+                               states={'draft': [('readonly', False)]})
+    quality_status = fields.Selection(selection=INDICATOR_STATUS,
+                                      readonly=True,
+                                      states={'draft': [('readonly', False)]})
+    quality_color = fields.Char(readonly=True,
+                                default='#00ff00',
+                                states={'draft': [('readonly', False)]})
+    quality_remarks = fields.Html(readonly=True,
+                                  states={'draft': [('readonly', False)]})
+    delay_status = fields.Selection(selection=INDICATOR_STATUS,
+                                    readonly=True,
+                                    states={'draft': [('readonly', False)]})
+    delay_color = fields.Char(readonly=True,
+                              default='#00ff00',
+                              states={'draft': [('readonly', False)]})
+    delay_remarks = fields.Html(readonly=True,
+                                states={'draft': [('readonly', False)]})
+    global_status = fields.Selection(selection=INDICATOR_STATUS,
+                                     readonly=True,
+                                     states={'draft': [('readonly', False)]})
+    global_color = fields.Char(readonly=True,
+                               default='#00ff00',
+                               states={'draft': [('readonly', False)]})
+    global_remarks = fields.Html(readonly=True,
+                                 states={'draft': [('readonly', False)]})
+    risks = fields.Html(readonly=True,
+                        states={'draft': [('readonly', False)]})
+    user_id = fields.Many2one(comodel_name='res.users',
+                              related='project_id.user_id',
+                              store=True,
+                              readonly=True)
+    indicator_ids = fields.One2many('project.status.indicator.value',
+                                    'report_id',
+                                    readonly=True,
+                                    states={'draft': [('readonly', False)]})
+    task_snapshot_ids = fields.One2many(
+        'project.task.snapshot',
+        'report_id',
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+
+    @api.multi
+    def action_confirm(self):
+        self.state = 'ready'
+
+    @api.multi
+    def action_validate(self):
+        self.state = 'published'
+
+    @api.model
+    def compute_indicator_values(self, project, date):
+        indicators = self.env['project.status.indicator.value'].browse()
+        for indicator in self.env['project.status.indicator'].search([]):
+            indicators |= indicator.compute_value(project, date)
```

Comment:
this could be slow with a lot of records. It's ok if you don't expect that much.